internal/manifests/targetallocator/adapters/config_to_prom_config.go

```diff
@@ -0,0 +1,283 @@
+			if !rcErr {
+				continue
+			}
+
+			_, rcErr = replacementProperty.(string)
+			if !rcErr {
+				return nil, errorNotAStringAtIndex("replacement", i)
+			}
+		}
+
+		metricRelabelConfigsProperty, ok := scrapeConfig["metric_relabel_configs"]
+		if !ok {
+			continue
+		}
+
+		metricRelabelConfigs, ok := metricRelabelConfigsProperty.([]interface{})
+		if !ok {
+			return nil, errorNotAListAtIndex("metric_relabel_configs", i)
+		}
+
+		for i, rc := range metricRelabelConfigs {
+			relabelConfig, ok := rc.(map[interface{}]interface{})
+			if !ok {
+				return nil, errorNotAMapAtIndex("metric_relabel_config", i)
+			}
+
+			replacementProperty, ok := relabelConfig["replacement"]
+			if !ok {
+				continue
+			}
+
+			_, ok = replacementProperty.(string)
+			if !ok {
+				return nil, errorNotAStringAtIndex("replacement", i)
+			}
+		}
+	}
+
+	return prometheus, nil
+}
+
+// AddHTTPSDConfigToPromConfig adds HTTP SD (Service Discovery) configuration to the Prometheus configuration.
+// This function removes any existing service discovery configurations (e.g., `sd_configs`, `dns_sd_configs`, `file_sd_configs`, etc.)
+// from the `scrape_configs` section and adds a single `http_sd_configs` configuration.
+// The `http_sd_configs` points to the TA (Target Allocator) endpoint that provides the list of targets for the given job.
+func AddHTTPSDConfigToPromConfig(prometheus map[interface{}]interface{}, taServiceName string) (map[interface{}]interface{}, error) {
+	prometheusConfigProperty, ok := prometheus["config"]
+	if !ok {
+		return nil, errorNoComponent("prometheusConfig")
+	}
+
+	prometheusConfig, ok := prometheusConfigProperty.(map[interface{}]interface{})
+	if !ok {
+		return nil, errorNotAMap("prometheusConfig")
+	}
+
+	scrapeConfigsProperty, ok := prometheusConfig["scrape_configs"]
+	if !ok {
+		return nil, errorNoComponent("scrape_configs")
+	}
+
+	scrapeConfigs, ok := scrapeConfigsProperty.([]interface{})
+	if !ok {
+		return nil, errorNotAList("scrape_configs")
+	}
+
+	sdRegex := regexp.MustCompile(`^.*(sd|static)_configs$`)
+
+	for i, config := range scrapeConfigs {
+		scrapeConfig, ok := config.(map[interface{}]interface{})
+		if !ok {
+			return nil, errorNotAMapAtIndex("scrape_config", i)
+		}
+
+		// Check for other types of service discovery configs (e.g. dns_sd_configs, file_sd_configs, etc.)
+		for key := range scrapeConfig {
+			keyStr, keyErr := key.(string)
+			if !keyErr {
+				continue
+			}
+			if sdRegex.MatchString(keyStr) {
+				delete(scrapeConfig, key)
+			}
+		}
+
+		jobNameProperty, ok := scrapeConfig["job_name"]
+		if !ok {
+			return nil, errorNotAStringAtIndex("job_name", i)
+		}
+
+		jobName, ok := jobNameProperty.(string)
+		if !ok {
+			return nil, errorNotAStringAtIndex("job_name is not a string", i)
+		}
+
+		escapedJob := url.QueryEscape(jobName)
+		scrapeConfig["http_sd_configs"] = []interface{}{
+			map[string]interface{}{
+				"url": fmt.Sprintf("http://%s:80/jobs/%s/targets", taServiceName, escapedJob),
```

Comment:
nit: We can also make this a global variable to be used in sync with `internal/manifest/service.go` just so if we decide to change it will be easier to change